### ISSUE TITLE
Daisy descriptor does not work through DescriptorExtractor/FeatureDetector

### ISSUE THREAD

**Issue by rasmusramsboel** [kyamagu/mexopencv#287]
In the DescriptorExtractor file the following is in the header:

% detector = cv.FeatureDetector('ORB');
% keypoints = detector.detect(img);
% extractor = cv.DescriptorExtractor('ORB');
% descriptors = extractor.compute(img, keypoints);

on how to extract descriptors. However the FeatureDetector does not work with Daisy as input, so the header is not genereric for all descriptors.

Opencv 3.1 and mexopencv 3.1

Still very nice to have a link of opencv and matlab :-)

**Comment by amroamroamro** [comment]
DAISY is not a feature detector, it is a class that computes descriptors for a set of keypoints.
Meaning you have to first detect keypoints using one of the supported detectors and then create a Daisy DescriptorExtractor to compute descriptors for every point detected. So it goes something like this:

``` matlab
detector = cv.FeatureDetector('..');   % choose a supported detector here
kpts = detector.detect(img);
extractor = cv.DescriptorExtractor('DAISY');
descs = extractor.compute(img, kpts);
```

You can either use the generic interfaces `cv.FeatureDetector`/`cv.DescriptorExtractor`, or directly use the classes in question `cv.ORB`, `cv.DAISY`, etc..

Notice that some classes can be detectors only (e.g `cv.FastFeatureDetector` only exposes a "detect" method), extractors only (e.g `cv.DAISY` only has a "compute" method), or both detectors and extractors at the same time (e.g `cv.ORB` provides a "detectAndCompute" method).

**Comment by amroamroamro** [comment]
I should also mention that DAISY was designed as a dense descriptor extractor, i.e computing descriptors of all pixels of an image. There's a specialized method in the cv.DAISY class for that: ["compute_all"](http://kyamagu.github.io/mexopencv/matlab/DAISY.compute_all.html).

Unfortunately last time I tested this there was apparently a bug in OpenCV that caused MATLAB to crash when `compute_all` is called... I didn't investigate this further as the problem likely lies on the OpenCV side not mexopencv.

**Comment by rasmusramsboel** [comment]
Man, you are fast, thanks for your reply :-)

I just figured that the code would be generic for all descriptors, and was dreaming of some experiment where I had to do very little myself :-)

Thanks a bunch again :-)

I encountered a problem when building using "opencv_contrib-3.1.0".  "bioinspired" produced an error so in order to solve it, I took and used "bioinspired" from the master rather than the release.

**Comment by amroamroamro** [comment]
Do you mean an error when building mexopencv or when building OpenCV itself?

I ask because I reported a [bug](https://github.com/opencv/opencv/issues/6370) in OpenCV a while ago which manifests in "bioinspired" when compiling opencv+opencv_contrib 3.1.0 using CMake >= 3.5.

If that's the case, I did submit a [workaround](https://github.com/opencv/opencv_contrib/issues/627) for that which was merged post 3.1.0, but you can also just downgrade to CMake 3.4 for now.

**Comment by rasmusramsboel** [comment]
Ok, I used the latest CMake 3.6, so maybe that was the same problem.

I'm not positive whether it was when building mexopencv or OpenCV but it might have been OpenCV, which would make sense according to your bug reporting. I remember something '*retina*' popping up, so very likely it's that and also that it is solvable by including the latest master for "bioinspired", where your bugfix has been added.
